internal/types/path.go

```diff
@@ -17,17 +17,81 @@ package types
 import (
 	"fmt"
 	"strconv"
+	"strings"
 )
 
+// Path represents field path type. It should be used wherever we work with paths or dot notation.
+type Path struct {
+	s []string
+}
```

Comment:
I think we should add `String() string` method that would return a dot-separated string